app/src/main/java/com/scouts/kitchenplaner/model/usecases/EditMealPlan.kt

```diff
@@ -51,27 +81,68 @@ class EditMealPlan @Inject constructor(
         )
     }
 
+    /**
+     * Removes the main recipe from a meal slot
+     *
+     * @param project Project which contains the meal slot
+     * @param mealSlot Meal slot from which the main recipe gets deleted
+     */
     suspend fun removeMainRecipeFromMeal(project: Project, mealSlot: MealSlot) {
         recipeManagementRepository.removeMainRecipeFromMealSlot(project.id, mealSlot)
     }
 
-
+    /**
+     * Removes all recipes from a meal slot
+     *
+     * @param  project Project which contains the meal slot
+     * @param mealSlot Meal slot from which all recipes get removes
+     */
     suspend fun removeRecipesFromMealSlot(project: Project, mealSlot: MealSlot) {
         recipeManagementRepository.removeRecipesFromMeal(project.id, mealSlot)
     }
 
+    /**
+     * Swaps all recipes from the first and the second meal slot
+     *
+     * @param project Project that contains both meal slots
+     * @param first First meal slot to swap all recipes
+     * @param second Second meal slot to swap all recipes
+     */
     suspend fun swapMealSlots(project: Project, first: MealSlot, second: MealSlot) {
         recipeManagementRepository.swapRecipes(project.id, first, second)
     }
 
+    /**
+     * Adds a new meal to the meal plan
+     *
+     * @param project Project to which a new meal is added
+     * @param meal Name of the new meal
+     * @param index Identifying when the meal happens w.r.t. the other meals already present
+     */
     suspend fun addMeal(project: Project, meal: String, index: Int = project.meals.size) {
         projectRepository.addMealToProject(meal, index, project.id)
     }
 
+    /**
+     * Removes a meal from a project.
```

Comment:
Are we adding meals to projects or the meal plan? Either is fine, it should be consistent between ```addMeal``` and ```removeMeal```, however.